src/components/questionnaire/questionnaire-table.tsx

```diff
@@ -34,9 +34,7 @@ const QuestionnaireTable = () => {
   const adminMode = UserRoleUtils.adminMode();
   const navigate = useNavigate();
   const [_, setMode] = useState<QuestionnairePreviewMode>();
-  const { questionnairesApi } = useLambdasApi();
-  const [questionnaires, setQuestionnaires] = useState<Questionnaire[]>([]);
-  const [loading, setLoading] = useState(false);
+  const { loading, setLoading, questionnaires, setQuestionnaires, questionnairesApi } = useFetchQuestionnaires();
```

Comment:
The creation of the hook is fine to reduce duplication, but I think returning the questionnairesApi from it is a little odd. I'd suggest not returning it from the hook and just accessing here in this component as you would normally.